addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js

```diff
@@ -71,22 +81,26 @@ class DataEgressService extends Service {
     const folderName = `${environment.id}/`;
 
     try {
-      s3Service.createFolder(bucketName, folderName);
+      s3Service.createPath(bucketName, folderName);
     } catch (error) {
       throw this.boom.badRequest(`Error in creating egress store:${folderName} in bucket: ${bucketName}`, true);
     }
 
     // prepare info for ddb and update egress store info
     const egressStoreId = uuid();
+    const creationTime = new Date().toISOString;
     const dbObject = {
       id: egressStoreId,
       egressStoreName: `${environment.name}-egress-store`,
+      createdAt: creationTime,
       createdBy: environment.createdBy,
       workspaceId: environment.id,
       projectId: environment.projectId,
       s3BucketName: bucketName,
       s3BucketPath: folderName,
-      status: 'Created',
+      status: CREATED_STATUS_CODE,
```

Comment:
somewhere we should disallow creating of the egress store if the entry already exists for the egress store.

Reply:
Updated in commit: 1076076507c38a7e39046b29d7fdacae3f50ed21